### PR TITLE
Added partial decomposition of conv_backward and grad_bias computation

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #89171
* #89203
* __->__ #89128

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

`convolution_backward` often just kicks off the `sum` as a separate kernel. Splitting it off in a decomp allows us to fuse it into other ops: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/Convolution.cpp#L2150

Improves `convnext_base` from 373 img/s => 383 img/s

Not sure what other models use convolution with bias haha.

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: torch/_inductor/decomposition.py [hunk 1]
```diff
@@ -148,6 +148,42 @@ def addmm(input, mat1, mat2, *, beta=1, alpha=1):
...
+    transposed,
+    output_padding,
+    groups,
+    output_mask,
+):
+    if not output_mask[2] or grad_output.device.type != "cuda":
+        return NotImplemented
+
+    grad_bias = aten.sum(
+        grad_output, [0, 2, 3, 4] if grad_output.dim() == 5 else [0, 2, 3]
```
Comment (ngimel) [inline torch/_inductor/decomposition.py:169]: `convolution_backward` can also have 3d inputs https://github.com/pytorch/pytorch/blob/7f55db4fb0fb12ed593c7f23de01bfb9330b7dd5/aten/src/ATen/native/Convolution.cpp#L1947-L1954

Comment (ngimel) [reply torch/_inductor/decomposition.py:169]: Luckily, we handle inputs without batch dimension before we come to `convolution_backward`, so 3d input is always 1d spatial

File: test/inductor/test_torchinductor.py [hunk 2]
```diff
@@ -68,6 +68,7 @@
 requires_cuda = functools.partial(unittest.skipIf, not HAS_CUDA, "requires cuda")
 
 torch._inductor.config.triton.autotune = False  # too slow
+torch.backends.cudnn.allow_tf32 = False
```
[the remark this answers is not in the document]
Comment (Chillee) [reply test/inductor/test_torchinductor.py]: I changed it to be local to the test.

File: test/inductor/test_torchinductor.py [hunk 3]
```diff
@@ -4177,6 +4177,78 @@ def fn(x):
...
+        weight = torch.randn(C, C, 3, 3, 3)
+
+        def shrink_rank(x, rank):
+            res = x
+            while res.dim() > rank:
+                res = torch.select(res, -1, 0)
+            return res.contiguous()
+
+        rank4_inps = [shrink_rank(x, 4) for x in [grad_out, inp, weight]]
+        rank3_inps = [shrink_rank(x, 4) for x in [grad_out, inp, weight]]
```
Comment (ngimel) [inline test/inductor/test_torchinductor.py:4243]: you meant 3 here?

**Pull request closed without merging**